balancer/rls/internal/keys/builder_test.go

```diff
@@ -160,14 +153,12 @@ func TestMakeBuilderMap(t *testing.T) {
 	for _, test := range tests {
 		t.Run(test.desc, func(t *testing.T) {
 			builderMap, err := MakeBuilderMap(test.cfg)
-			if (err != nil) != test.wantErr {
-				t.Errorf("MakeBuilderMap(%+v) returned err: %v, want: %v", test.cfg, err, test.wantErr)
+			if (err != nil) != (test.wantErrPrefix != "") {
+				t.Errorf("MakeBuilderMap(%+v) returned err: %v, want: %v", test.cfg, err, test.wantErrPrefix)
 			}
```

Comment:
You could remove this `if` by changing the one below slightly:
```
if (test.wantErrPrefix != "" && !strings.HasPrefix(err.Error(), test.wantErrPrefix)) ||
   (test.wantErrPrefix == "" && err != nil) ||
   !cmp.Equal(builderMap, test.wantBuilderMap, cmp.AllowUnexported(builder{}, matcher{})) {
```

Reply:
Won't the call to `err.Error` panic when `test.wantErrPrefix` is non-empty, but `err == nil`

Reply:
Oops, yes.... anyway, there are a few ways you can get it down to a single `if`, which would be nicer to avoid the two `t.Error`s

Reply:
Done. Split the test into two, one for the success case and one for the error case. This seems to be preferred way based on internal Go unit testing codelab. Thanks.

Reply:
OK, that sounds fine too, as long as the duplication between the two isn't too much.  Often it's different enough that it's not worth sharing.